packages/m/mkl/xmake.lua

```diff
@@ -109,7 +113,8 @@ package("mkl")
         package:add("links", package:is_arch("x64", "x86_64") and "mkl_blas95_" .. suffix or "mkl_blas95")
         package:add("links", package:is_arch("x64", "x86_64") and "mkl_lapack95_" .. suffix or "mkl_lapack95")
 
-        if package:has_tool("cc", "gcc", "gxx") then
+        if (package:has_tool("cc", "gcc", "gxx") or 
+            package:has_tool("fc", "gfortran")) then
```

Comment:
Here we should check `package:has_tool("cc", "cl", "clang-cl")` first, otherwise on windows if there is gfortran installed it will lead to incorrect link flags

Reply:
Basically it should also detect Apple clang since Apple clang do not support link groups. But I don't know how to tell Apple clang from llvm clang using package:has_tool() api. @waruqi any idea?

Reply:
use package:has_flags() instead of has_tool?

Reply:
you can try, test linker will be better.

Reply:
fcldflags (binary) fcshflags (shared)

Reply:
https://github.com/xmake-io/xmake-repo/blob/f008565591c68b4d954d779b4dda4647cc0da4bc/packages/o/openmp/xmake.lua#L76-L78

Reply:
It seems the new test has improved the results of the tests (in the meantime I discovered I did not need to load add_configs for `fetch.lua` for the 'runtime' test to work), but it still fails on Mac OS X x86-64.
Is it possible that the LLVM linker on Apple is also recognized as GCC?
Should I replace the test:
```if (package:has_tool(toolkind, "gcc", "gxx") or package:has_tool(toolkind, "gfortran")) then```
with something of the form:
```
if (not package:has_tool(toolkind, "clang", "clangxx") and (package:has_tool(toolkind, "gcc", "gxx") or package:has_tool(toolkind, "gfortran"))) then
```
?

For Fortran, I had to replace `fcldflags` with `ldflags` since the support of the former seems broken up to xmake *xmake v2.9.5+dev.b938556*.  Using `ldflags` is necessary to compile Fortran programs with GFortran on Linux AMD64.  I have commented out the `fcldflags` and `fcshflags` for testing purposes in the package source.